### DISCUSSION TITLE
Messed up ContentType for Attachments

### DISCUSSION THREAD

**Discussion by mcc-github-123456789** [ravendb/ravendb#16870] [Q&A] [answered]
RandomDocuments/59663-A
{   
    "NameOnDocument": "John Doe",
    "AcceptableForReview": null,
    "IsLocked": null,
    "CreatedBy": "system",
    "CreatedDate": "2023-04-10T10:57:29.0000000",
    "LastModifiedBy": "imbored",
    "LastModifiedDate": "2023-05-25T19:23:49.0000000",
    "@metadata": {
        "@collection": "RandomDocuments",
        "Raven-Clr-Type": "Abc.ContentTypes.RandomDocuments, Abc.ContentTypes",
        "@flags": "HasRevisions, FromReplication, HasAttachments",
        "@id": "RandomDocuments/59663-A",
        "@last-modified": "2023-05-26T15:00:24.5968119Z",
        "@attachments": [
            {
                "Name": "bddc623e-83a0-449d-95a4-f3ff64fa25aa.pdf",
                "Hash": "BAWdGke7t02mo6LfwxhJ4oJ0aI9d0wAVoj/vk2+vkJM=",
                "ContentType": "text/plain",
                "Size": 948272
            },
            {
                "Name": "bddc623e-83a0-449d-95a4-f3ff64fa25aa.txt",
                "Hash": "4plHj8qGwBsYqTOy6g6Q4buy6OVkXbSuT1jsBucLLGg=",
                "ContentType": "application/pdf",
                "Size": 209601
            }
        ],
        "@change-vector": "B:23323394-6EQ4qChW8EOoeBKyKHXRWA, A:23334973-KGgUoXIJ4EiaXyGnGxFrlw, C:23318013-f2/NWlORBkCJbW9pL2N0qw",
        "@index-score": 1.533555030822754
    }
}
 
On the migration of some documents, we ended up with data that looks like the above. Note that the ContentType doesn't match the attachment name extension. How would I write an RQL query to iterate through the attachments and find all the offending documents in our database? I can write basic RQL queries but one that iterates through the attachments collection appears to be a bridge too far for me. Spinning my wheels. Second to that, how do I go about writing a patch that will correct the ContentType so that it corresponds to the extension in the name for all documents in our DB?
I really appreciate any help you can provide.
Adam

**Comment by ppekrol** [comment, marked as answer]
Hi, inside patch you cannot modify the attachments, but you can create a static index that indexes the attachment metadata and query that. Check here: https://ravendb.net/docs/article-page/5.4/csharp/document-extensions/attachments/indexing
